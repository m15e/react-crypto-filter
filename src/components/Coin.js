import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import numeral from 'numeral';

const Coin = props => {
  const {
    id, name, price, marketCap, allTimeHigh, img,
  } = props;

  const price$ = numeral(price).format('$0,0.00');
  const mrktCap$ = numeral(marketCap).format('$0,0.00');
  const allTime$ = numeral(allTimeHigh).format('$0,0.00');

  return (
    <div className="coin tile is-child box is-4" id={id}>
      <figure className="image is-64x64 coin-logo">
        <img className="is-rounded" src={img} alt={name} />
      </figure>
      <h3 className="title is-5">{name}</h3>
      <p>
        <span>Market Capitalization</span>
        <br />
        {mrktCap$}
      </p>
      <p>
        <span>Current Price</span>
        <br />
        {price$}
      </p>
      <p>
        <span>All Time High</span>
        <br />
        {allTime$}
      </p>
      <Link to={`/coin/${id}`} className="view-coin button is-ghost is-small">
        View Coin
      </Link>
    </div>
  );
};

Coin.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  allTimeHigh: PropTypes.number.isRequired,
  img: PropTypes.string.isRequired,
  marketCap: PropTypes.number.isRequired,
};

export default Coin;
