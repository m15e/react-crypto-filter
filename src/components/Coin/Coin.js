import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import numeral from 'numeral';

const Coin = props => {
  const {
    id, name, price, marketCap, allTimeHigh, img,
  } = props;

  const price$ = numeral(price).format('$0,0.00');
  const mrktCap$ = numeral(marketCap).format('$0,0.00');
  const allTime$ = numeral(allTimeHigh).format('$0,0.00');

  return (
    <div className="coin tile is-child box is-4" id={id}>
      <figure className="image is-64x64 coin-logo">
        <img className="is-rounded" src={img} alt={name} />
      </figure>
      <h3 className="title is-5">{name}</h3>
      <p>
        <span>Market Capitalization</span>
        <br />
        {mrktCap$}
      </p>
      <p>
        <span>Current Price</span>
        <br />
        {price$}
      </p>
      <p>
        <span>All Time High</span>
        <br />
        {allTime$}
      </p>
      <Link to={`/coin/${id}`} className="view-coin button is-ghost is-small">
        View Coin
      </Link>
    </div>
  );
};

Coin.defaultProps = {
  id: 'coinId',
  name: 'coin',
  price: 1,
  allTimeHigh: 1,
  img: 'https://assets.coingecko.com/coins/images/1/large/bitcoin.png',
  marketCap: 1,
};

Coin.propTypes = {
  id: PropTypes.string,
  name: PropTypes.string,
  price: PropTypes.number,
  allTimeHigh: PropTypes.number,
  img: PropTypes.string,
  marketCap: PropTypes.number,
};

export default Coin;
