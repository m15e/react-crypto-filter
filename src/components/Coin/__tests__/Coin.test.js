import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Coin from '../Coin';

it('Coin component matches snapshot', () => {
  const tree = renderer.create(<BrowserRouter><Coin /></BrowserRouter>).toJSON();
  expect(tree).toMatchSnapshot();
});