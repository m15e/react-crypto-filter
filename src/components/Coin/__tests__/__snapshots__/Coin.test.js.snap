// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coin component matches snapshot 1`] = `
<div
  className="coin tile is-child box is-4"
  id="coinId"
>
  <figure
    className="image is-64x64 coin-logo"
  >
    <img
      alt="coin"
      className="is-rounded"
      src="https://assets.coingecko.com/coins/images/1/large/bitcoin.png"
    />
  </figure>
  <h3
    className="title is-5"
  >
    coin
  </h3>
  <p>
    <span>
      Market Capitalization
    </span>
    <br />
    $1.00
  </p>
  <p>
    <span>
      Current Price
    </span>
    <br />
    $1.00
  </p>
  <p>
    <span>
      All Time High
    </span>
    <br />
    $1.00
  </p>
  <a
    className="view-coin button is-ghost is-small"
    href="/coin/coinId"
    onClick={[Function]}
  >
    View Coin
  </a>
</div>
`;
